code/modules/economy/ATM.dm

```diff
@@ -76,9 +76,39 @@ log transactions
 			to_chat(user, SPAN_INFO("You insert [I] into [src]."))
 			src.attack_hand(user)
 			qdel(I)
+	else if(istype(I, /obj/item/holder/cat) || istype(I, /obj/item/holder/Jones))
+		user.visible_message(SPAN_DANGER("[user] begins stuffing [I] into the ATM!"))
+		playsound(src, "sound/machines/fax.ogg", 5)
+		if(!do_after(user, 70, INTERRUPT_ALL, BUSY_ICON_BUILD))
+			return
+		visible_message(SPAN_DANGER("You hear a loud metallic grinding sound."))
+		playsound(src, 'sound/effects/splat.ogg', 25, 1)
+		playsound(src, "sound/voice/meow.ogg", 15)
+
+		for(var/mob/M in I.contents)
+
+			if(M.client) // if someone was playing the mob, log it
+				M.attack_log += "\[[time_stamp()]\] Was gibbed by <b>[key_name(user)]</b>"
+				user.attack_log += "\[[time_stamp()]\] Gibbed <b>[key_name(M)]</b>"
+				msg_admin_attack("[key_name(user)] gibbed [key_name(M)] in [user.loc.name] ([user.x], [user.y], [user.z]).", user.x, user.y, user.z)
+
+			M.spawn_gibs()
+			M.death(create_cause_data("ATM", user), TRUE)
```

Comment:
might runtime due to cause data being a string instead of datum

Reply:
Did not appear to runtime in testing, please confirm.